src/reflect/scala/reflect/internal/Names.scala

```diff
@@ -41,16 +41,19 @@ trait Names extends api.Names {
   private val typeHashtable = new Array[TypeName](HASH_SIZE)
 
   /**
-   * The hashcode of a name depends on the first, the last and the middle character,
+   * The hashcode of is equivalent to cs name depends on the first, the last and the middle character,
    * and the length of the name.
    */
-  private def hashValue(cs: Array[Char], offset: Int, len: Int): Int =
-    if (len > 0)
-      (len * (41 * 41 * 41) +
-       cs(offset) * (41 * 41) +
-       cs(offset + len - 1) * 41 +
-       cs(offset + (len >> 1)))
-    else 0
+  private def hashValue(cs: Array[Char], offset: Int, len: Int): Int = {
```

Comment:
This looks to me exactly the same as in scala https://github.com/scala/scala/blob/2.12.x/src/reflect/scala/reflect/internal/Names.scala#L233, so I still don't understand where the difference is to dotty..